src/tools/replay.cpp

```diff
@@ -86,6 +86,9 @@ bool has_next_action() {
 }
 
 std::string next_action_type() {
+	if(next_action == nullptr){
+		throw "Replay error! No action left to check type";
```

Comment:
Please throw a `std::string` instead of a `const char*`, since the last-chance catch-all in `main` only handles the former.

That applies both here and in `pop_next_action()`.